cypress/e2e/spec.cy.js

```diff
@@ -0,0 +1,92 @@
+describe('initial', () => {
+  it('passes', () => {
+    cy.visit('http://localhost:8080/')
+  })
+})
+
+describe('Chart Builder', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:8080/line.html');
+    
+      // Filling in label inputs
+      cy.get('#chart-title-input').type('Cats vs. Dogs');
+      cy.get('#chart-color-input').invoke('val', '#ff0000').trigger('input');
+      cy.get('#x-label-input').type('Cats');
+      cy.get('#y-label-input').type('Dogs');
+  
+      // Add values
+      for(let i = 0; i < 5; i++) {
+        cy.get('#add-values-btn').click();
+      }
+      cy.get('#x-y-data-grid').find('input').eq(2).type('1');
+      cy.get('#x-y-data-grid').find('input').eq(3).type('3');
+      cy.get('#x-y-data-grid').find('input').eq(4).type('2');
+      cy.get('#x-y-data-grid').find('input').eq(5).type('7');
+      cy.get('#x-y-data-grid').find('input').eq(6).type('3');
+      cy.get('#x-y-data-grid').find('input').eq(7).type('15');
+      cy.get('#x-y-data-grid').find('input').eq(8).type('4');
+      cy.get('#x-y-data-grid').find('input').eq(9).type('25');
+      cy.get('#x-y-data-grid').find('input').eq(10).type('5');
+      cy.get('#x-y-data-grid').find('input').eq(11).type('40');
+      cy.get('#x-y-data-grid').find('input').eq(12).click();
+  });
+
+  it('clears chart correctly', () => {
+    
+    cy.get('#generate-chart-btn').click();
+    cy.get('#clear-chart-btn').click();
+
+    // Assertion
+    cy.get('#chart-title-input').should('have.value', '');
+    cy.get('#x-y-data-grid').find('input').should('have.length', 4); // cleared
+  });
+
+  it('generation', () => {
+
+    // Assertion
+    cy.get('#generate-chart-btn').click();
+    cy.get('#chart-display').should('not.be.empty');
+
+    cy.get('#x-y-data-grid').find('input').should('have.length', 14);
```

Comment:
can you explain this?

Reply:
the additon of comments would be helpful. i don't know if it was in the requirement for the final but i think well documented was something we where scored on for all the other projects